test/unit/rooseveltUglifyTest.js

```diff
@@ -0,0 +1,227 @@
+        compiler: {
+          nodeModule: '../../roosevelt-uglify',
+          showWarnings: true,
+          params: {
+          }
+        }
+      }
+    }, gOptions)
+
+    // fork the app and run it as a child process
+    const testApp = fork(path.join(appDir, 'app.js'), {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    // an error should be thrown by the testApp, with a warnings in the string
+    testApp.stderr.on('data', (data) => {
+      if (data.toString().includes('Warnings')) {
+        testApp.kill()
+        done()
+      }
+    })
+
+    // It should not be able to complete initialization, meaning if it does, we have an error in the error handling
+    testApp.on('message', (params) => {
+      assert.fail('app was able to complete initialize and did not throw a warnings error')
+      testApp.kill()
+      done()
+    })
+  })
+
+  it('should not give a "warning" string since the showWarning param is false', function (done) {
+    // generate the app
+    generateTestApp({
+      appDir: appDir,
+      generateFolderStructure: true,
+      js: {
+        compiler: {
+          nodeModule: '../../roosevelt-uglify',
+          showWarnings: false,
+          params: {
+          }
+        }
+      }
+    }, gOptions)
+
+    // fork the app and run it as a child process
+    const testApp = fork(path.join(appDir, 'app.js'), {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    // an error should not be thrown by the testApp
+    testApp.stderr.on('data', (data) => {
+      if (data.toString().includes('Warnings')) {
+        assert.fail('app had thrown an error when showWarnings was set to false')
+        testApp.kill()
+        done()
+      }
+    })
+
+    // It should be able to complete initialization, meaning that the test had succeeded if it has completed initialization
+    testApp.on('message', (params) => {
+      testApp.kill()
+      done()
+    })
+  })
+
+  it('should give a "error" string if there is a massive problem with the code that the program is trying to parse (typo)', function (done) {
+    // JS source script that has a error in it (typo)
+    const errorTest = `function f(){ returbn 2 + 3; }`
+    // path of where the file with this script will be located
+    const pathOfErrorStaticJS = path.join(appDir, 'statics', 'js', 'b.js')
+    // make this file before the test
+    fs.writeFileSync(pathOfErrorStaticJS, errorTest)
+
+    // generate the app
+    generateTestApp({
+      appDir: appDir,
+      generateFolderStructure: true,
+      js: {
+        compiler: {
+          nodeModule: '../../roosevelt-uglify',
+          showWarnings: false,
+          params: {
+          }
+        }
+      }
+    }, gOptions)
+
+    // fork the app and run it as a child process
+    const testApp = fork(path.join(appDir, 'app.js'), {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    // an error should be thrown by the testApp
+    testApp.stderr.on('data', (data) => {
+      if (data.toString().includes('failed')) {
```

Comment:
Could you extend this check to include more words, like `failed to parse`, failed is pretty broad and could happen from many errors.